docs/Theory_cluster_batch_scheduling.rst

```diff
@@ -1,15 +1,132 @@
+-----------------------
+
+The available resources on the cluster are not infinite and it is the
+workload manager's job to allocate them. Whenever a job request comes
+in and there are not enough resources available to start it
+immediately, it will go in the queue.
+
+Once a job is in the queue, it will stay there until another job
+finishes and then the workload manager will try to use the newly freed
+resources with jobs from the queue. The exact order in which the jobs
+will start is not fixed, because it depends on the local policies
+which can take into account the user priority, the time since the job
+was requested, the amount of resources requested and possibly other
+things. There should be a tool that comes with the manager where you
+can see the status of your queued jobs and why they remain in the
+queue.
+
+
+About partitions
+----------------
+
+The workload manger will divide the cluster into partitions according
+to the configuration set by the admins. A partition is a set of
+machines typically reserved for a particular purpose. An example might
+be CPU-only machines for preprocessing setup as a separate partition.
+It is possible for multiple partitions to share resources.
+
+There will always be at least one partition that is the default
+partition in which jobs without a specific request will go. Other
+partitions can be requested, but might be restricted to a group of
+users, depending on policy.
+
+Partitions are useful for a policy standpoint to ensure efficient use
+of the cluster resources and avoid using up too much of one resource
+type blocking use of another. They are also useful for heterogenous
+clusters where different hardware is mixed in and not all software is
+compatible with all of it (for example x86 and POWER cpus).
+
+
+Exceding limits (preemption and grace periods)
+----------------------------------------------
+
+To ensure a fair share of the computing resources for all the workload
+manager establishes limits on the amount of resources that a single
+user can use at once. These can be hard limits which prevent running
+jobs when you go over or soft limits which will let you run jobs, but
+only until some other job needs the resources.
+
+Admin policy will determine what those exact limits are for a
+particular cluster or user and whether they are hard or soft limits.
+
+They way soft limits are enforced is using preemption, which means
+that when another job needs the resources that your job is using your
+job will receive a singal that it needs to save its state and exit.
+It will be given a certain amount of time to do this (the grace
+period, which may be 0s) and then forcefully terminated if it is still
+running.
+
+Depending on the workload manager in use and the cluster configuration
+a job that is preempted like this may be automatically rescheduled to
+have a chance to finish or it may be up to the job to reschedule
+itself.
+
+The other limit you can encounter with a job that goes over job
```

Comment:
I'm not sure to understand this sentence. What do you mean?

Reply:
Check if the new wording is clearer, please.